scripts/bsd/openbsd-prep-postgres.sh

```diff
@@ -0,0 +1,36 @@
+#!/bin/sh
+
+pkg_add -I \
+    vim--no_x11 git \
+    bash \
+    git \
+    gmake \
+    meson \
+    pkgconf \
+    \
+    bison \
+    ccache \
+    gettext-tools \
+    \
+    p5-IPC-Run \
+    \
+    gssdp \
+    icu4c \
+    libxml \
+    libxslt \
+    lz4 \
+    openpam \
+    python%3.8 \
+    readline \
+    tcl%8.6 \
+    \
+    login_krb5 \
+    openldap-client--gssapi \
+    openldap-server--gssapi
+
+# Set kernel parameters for running postgres tests
+echo "/sbin/sysctl kern.seminfo.semmni=2048" >> /etc/rc.local
+echo "/sbin/sysctl kern.seminfo.semmns=32768" >> /etc/rc.local
+
+# Forbid root username/password login
+sed -i 's/PermitRootLogin yes/PermitRootLogin prohibit-password/g' /etc/ssh/sshd_config
```

Comment:
Doesn't this mean the password is left in the 'vanilla' images? We shouldn't need password auth once the image is uploaded to GCP, since the accounts / ssh keys should be managed by the GCP scripts, no?

Reply:
If 'we' want to connect the 'vanilla' images it works without passwords but packer needs to connect with username / password for creating 'postgres' images. We can add another step which is for creating second vanilla images from first vanilla images. We can remove password in these step, then we can remove first vanilla image but this is a bit costly.

Reply:
> If 'we' want to connect the 'vanilla' images it works without passwords but packer needs to connect with username / password for creating 'postgres' images

Why / When does packer need that? That step runs in gcp, and packer can create temporary ssh keys for use with gcp. That's how it works for the debian / freebsd images currently...

Reply:
These are the steps:

1. Create 'vanilla' VM from ISO file
2. Export that VM as a image file to the Google Images
    2.1. We have 'vanilla' image in Google Images now
3. Create and export 'postgres' image using 'vanilla' image

I think since we use vanilla VM's image(not raw image like in the GCP), packer can't connect the image without username and passwords. Packer can create temporary ssh keys but 'vanilla' image doesn't have public key of temporary ssh key so it can't connect with using ssh key pair.

Alternative idea: We can create 'postgres' image like we create 'vanilla' image. We can create it from ISO file, then no need to have a username / password.

Reply:
packer's ssh keys are supposed to be provided via the gcp infrastructure / scripts. If that doesn't work, something else is wrong. How exactly does it fail?

Reply:
Ping?

Reply:
This is what happens if I remove ssh's pasword 'postgres' image: https://cirrus-ci.com/task/4592026892107776

Startup script in these images work like that: search & download users' public keys to the user' folders. Then user need to pass private key to the ssh command. Since packer is not a user in our GCP, startup scripts won't download its ssh keys. I am not sure how packer provide its keys via the gcp infrastructure / scripts but if its something similar to metadata, then we can add that to startup scripts.

Reply:
I think it creates an ephemeral user. It might just be that it relies on `ssh_username            = "packer"` that the others use, i.e. that root doesn't work.

Reply:
Same error: https://cirrus-ci.com/task/4771415630020608

I can search details of how packer transfer its keys.

Reply:
We create a user like that, that's why it works for linux / freebsd. I suspect that there's some problem with the script handling the key installation (likely not handling the unix username differing from the GCP username), preventing this from working.  Did you try doing this interactively with packer build -debug, and logging into the *bsd machine to see what happens on the machine?

Reply:
Done.

Reply:
What did you end up having to change?